utils/rowgroup/rowgroup.cpp

```diff
@@ -1229,9 +1229,9 @@ uint32_t RowGroup::getDataSize() const
   return headerSize + (getRowCount() * offsets[columnCount]);
 }
```

Comment:
I think we should change `getDataSize()` too.
```c++
uin64_t RowGroup::getDataSize() const
{
  return getDataSize(getRowCount());
}
```

Reply:
Anyway, IMO there should be `static_cast<uint64_t>(offset_field)` in every `getDataSize`-like function to prevent int overflow when multiplying two uint32_t values.

Reply:
Only `getDataSize` this makes sense. B/c the other two has 8192 as a number of rows so there must be 65K columns in the rowgroup to actually overflow.

Reply:
We have a hard limit of 4096 columns per table. It will requre query like ```SELECT * FROM t1 JOIN t2 ON t1a = t2.b JOIN ...``` to join more than 16 tables  with max columns (or more than 32 - most values in row will occupy 8 bytes), to have an overflowiing condition.

But, we may hit another hard limit on row size on the MDB's side (online documentation has it at 64K).

What I'd suggest is to have a comment describing this situation, here or elsewhere.